scalanet/test/src/io/iohk/scalanet/peergroup/UDPPeerGroupSpec.scala

```diff
@@ -99,24 +103,18 @@ class UDPPeerGroupSpec extends FlatSpec with EitherValues {
       _ <- echoServer(pg2).startAndForget
       response <- requestResponse(pg1, pg2.processAddress, messageFromClients)
       response1 <- requestResponse(pg3, pg2.processAddress, messageFromClients)
-      numOfServerChannelsAfter2Requests = pg2.activeChannels.size()
       response2 <- requestResponse(pg1, pg2.processAddress, messageFromClients)
       response3 <- requestResponse(pg3, pg2.processAddress, messageFromClients)
-      numOfServerChannelsAfter4Requests = pg2.activeChannels.size()
-      _ = testScheduler.tick(pg2.config.cleanUpInitialDelay)
-      numOfServerChannelsAfterCleanup = pg2.activeChannels.size()
-      numOfClient1ChannelsAfterCleanup = pg1.activeChannels.size()
-      numOfClient2ChannelsAfterCleanup = pg3.activeChannels.size()
     } yield {
       response shouldEqual messageFromClients
       response1 shouldEqual messageFromClients
       response2 shouldEqual messageFromClients
       response3 shouldEqual messageFromClients
-      numOfServerChannelsAfter2Requests shouldEqual 2
-      numOfServerChannelsAfter4Requests shouldEqual 4
```

Comment:
currently there is no way to check when channels are cleared, as they are cleared asynchornously in background as soos as possbile by netty threads. Thats why only assertion we can get is:
```
      eventually {
        assert(pg1.activeChannels.isEmpty)
        assert(pg2.activeChannels.isEmpty)
        assert(pg3.activeChannels.isEmpty)
      }
```

Reply:
I see :thinking: and in fact the previous check was a bit of lie, right? We would check which channels were active but some of them could have already been close from the netty perspective.

Reply:
you can say that, closing our channel was slightly disconnected from closing netty channel for client channels.
On thing to remember, is that on server side there is only one netty channel and multliple our channels. So when we close server channel on our side, we do not close netty server channel as it would effectivly kill server.